### ISSUE TITLE
ipython 5.8.0 incompatible with Python 3.8

### ISSUE THREAD

**Issue by simonpintarelli** [aiidateam/aiida-core#3559]
It seems that for Python 3.8  `ipython>=7.3` is required: https://ipython.org/news.html#ipython-7-3

ipython 5.8.0 is broken with Python 3.8:

```
Python 3.8.0 (default, Oct 23 2019, 18:51:26) 
Type "copyright", "credits" or "license" for more information.

IPython 5.8.0 -- An enhanced Interactive Python.
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
/usr/lib64/python3.8/codeop.py in __call__(self, source, filename, symbol)
    131 
    132     def __call__(self, source, filename, symbol):
--> 133         codeob = compile(source, filename, symbol, self.flags, 1)
    134         for feature in _features:
    135             if codeob.co_flags & feature.compiler_flag:

TypeError: required field "type_ignores" missing from Module
```

**Comment by sphuber** [comment]
This requirement currently comes because `ipython>=6` requires `pyzmq>=17` but `circus` which runs our daemon does not support this yet. I have created PR to circus which makes it compatible so now we just need to wait for a release. Once that is done we can relax the requirement of ipython as well.

**Comment by TsumiNa** [comment]
@sphuber `circus` has released v0.16.0 with dependence `pyzmq >= 17`. Maybe you can update the dependence in aiida-core now.

https://github.com/circus-tent/circus/releases/tag/0.16.0

**Comment by sphuber** [comment]
Thanks for the heads up, but I know as I was the one that made it compatible :) I already created a branch for it as well: https://github.com/sphuber/aiida_core/tree/fix_3559_ipython_requirement
However, tests are failing because it first needs to become available on conda-forge
